### PR TITLE
feat(core): create and open cloud workspace if not exists after logged in

### PR DESCRIPTION
**Pull request by CatsJuice**
Only execute when `﻿initCloud=true` is specified in the URL search params.

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#6511** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6511?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by CatsJuice — COMMENTED** [review]
@hwangdev97 cc
A trigger has been added, with the parameter ﻿`initCloud=true` specified in the URL.

File: packages/frontend/core/src/pages/index.tsx [hunk 1]
```diff
@@ -28,25 +38,52 @@ export const Component = () => {
...
     if (list.length === 0) {
       return;
     }
+    // check is user logged in && has cloud workspace
+    if (
+      searchParams.get('initCloud') === 'true' &&
+      status === 'authenticated'
+    ) {
+      searchParams.delete('initCloud');
+      if (list.every(w => w.flavour !== WorkspaceFlavour.AFFINE_CLOUD)) {
```
Comment (EYHN) [inline packages/frontend/core/src/pages/index.tsx:77]: maybe check `WorkspaceListService.status$.isLoading !== true` here

Comment (CatsJuice) [reply packages/frontend/core/src/pages/index.tsx:77]: In this case, I think we also need to wait for WorkspaceList loaded to open last opened workspace? Plz check the new logic.

**Review by EYHN — APPROVED** [review]

**Pull request merged**